ui/src/app/services/test-step.service.ts

```diff
@@ -63,6 +63,11 @@ export class TestStepService {
   }
 
   public create(step: TestStep): Observable<TestStep> {
+    if (step.addonTestData){
+    step.addonTemplate.parameters.filter(a=> a.allowedValues.length==0).forEach(p=>{
```

Comment:
is there any possibility of allowedValues === null?

Reply:
Yes, only for operator there are allowed values. For test-data variables there are not set of allowed values any value can be taken.